### PR TITLE
BAT-8249: Always fallback to prefix

### PR DESCRIPTION
**Pull request by olivierlefloch**
We want to be able to purge obsolete links, but for customers who have shortened links under their domain, we need to redirect to their domain page (workfor.us/company/er34t → workfor.us/company)

### CODE REVIEW

File: django_short_urls/views.py [hunk 1]
```diff
@@ -58,15 +57,17 @@ def main(request, path):
 
     link = Link.find_by_hash(path)
 
+    redirect_suffix = None
+
     if link is None:
-        # Try to find a matching short link by removing valid "catchall" suffixes
-        path_prefix, redirect_suffix = suffix_catchall.get_hash_from(path)
+        # Try to find a matching prefix
+        parts = path.rsplit('/', 1)
```
Comment (asiebert) [inline django_short_urls/views.py]: `rsplit` or `lsplit`? Do we ever create URLs with more than two dash-separated components?

Comment (olivierlefloch) [reply django_short_urls/views.py]: We can, yes, sometimes – at least for manually created urls – also, users could send broken stuff.

Comment (asiebert) [reply django_short_urls/views.py]: In that case, why would we try to find a prefix in the right-most slashed part?

Comment (olivierlefloch) [reply django_short_urls/views.py]: Why would we want to try the left most one? We should remove suffixes one by one.

Comment (asiebert) [reply django_short_urls/views.py]: It's not recursive though, we make a single attempt. To clarify my thoughts, this doesn't make sense to me:
```
path = "CompanySlug/AJobRef/garbage/junk"
parts = path.rsplit('/', 1)
# parts[0] == "CompanySlug/AJobRef/garbage"
Link.find_by_hash(parts[0])
```
Maybe that's an edge case (paths with >2 parts) we don't care about, but the use of `rsplit` is oddly specific, and I don't have a spec or test to validate it against :-)
